relay-event-normalization/src/normalize/mod.rs

```diff
@@ -479,6 +480,76 @@ fn normalize_measurements(
     }
 }
 
+/// Computes performance score measurements.
+///
+/// This computes score from vital measurements, using config options to define how it is
+/// calculated.
+fn normalize_performance_score(
+    event: &mut Event,
+    performance_score: Option<&PerformanceScoreConfig>,
+) {
+    let Some(performance_score) = performance_score else {
+        return;
+    };
+    for profile in &performance_score.profiles {
+        if let Some(condition) = &profile.condition {
+            if !condition.matches(event) {
+                continue;
+            }
+            if let Some(measurements) = event.measurements.value_mut() {
+                let mut should_add_total = false;
+                if !profile
+                    .score_components
+                    .iter()
+                    .all(|c| measurements.contains_key(c.measurement.as_str()))
+                {
+                    // Check all measurements exist, otherwise don't add any score components.
+                    break;
+                }
+                let mut score_total = 0.0;
+                for component in &profile.score_components {
+                    if let Some(value) = measurements.get_value(component.measurement.as_str()) {
+                        let subscore =
+                            utils::calculate_cdf_score(value, component.p10, component.p50)
+                                * component.weight;
+                        score_total += subscore;
+                        should_add_total = true;
+
+                        measurements.insert(
+                            format!("score.{}", component.measurement),
+                            Measurement {
+                                value: subscore.into(),
+                                unit: (MetricUnit::Fraction(FractionUnit::Ratio)).into(),
+                            }
+                            .into(),
+                        );
+                        measurements.insert(
+                            format!("score.weight.{}", component.measurement),
+                            Measurement {
+                                value: component.weight.into(),
+                                unit: (MetricUnit::Fraction(FractionUnit::Ratio)).into(),
+                            }
+                            .into(),
+                        );
```

Comment:
Do weights depend on the event? It seems these measurements are the same values as in the project config:
https://github.com/getsentry/sentry/pull/58418/files#diff-5e30755537fcfd5687d4399272065e49affcc12a4cf7a0bc377cff5234b8a08aR418-R421

Reply:
The weights can depend on the profile used to calculate the score, which depend on device type, and potentially browser etc. as well. To be able to graph the 'missed performance' each score needs to be compared to the aggregate weight. It's fine if we only have a single calculation, but additional calculations begin to make the weight logic complicated.

Reply:
eg. one event may have an FCP score of say 10 of 20, and another could use a different profile and have a score of 20 of 40. If you were to average scores it'd be 15, if you naively use weights it'd seem your score is 15 of 20 = 75%, but in reality it's 50% since those event have different evaluations.

Reply:
cc @edwardgou-sentry

Reply:
Depending how useful these metrics are for users to look directly at, we could also add them to the event's context.

Reply:
Hey @jjbayer and @iker-barriocanal, I don't think we'll need distributions for the weights. If we move the weights to `event.data`, will it be possible to query scores by weight tag value? I think that's mostly what we will be using these weights for, @k-fish please let me know if im wrong.

Reply:
Not sure I understand the question, it doesn't seem we're using tags here.

Am I correct to assume that the relevant data is the ratio? In the examples above, we're not interested in the 10 or 20, or 20 or 40; only that they have a 50% score. In that case, Relay could send a `0.5` representing the 50% instead of the raw values and weights. I also assume the weights won't change often.

Reply:
> will it be possible to query scores by weight tag value?

Not claiming I understand the use case 100%, but if we want to group by / filter to a specific value of the weight, would it make sense to add a metrics tag `"profile"` so we can query for example the avg. score for a specific profile?

Reply:
> Not claiming I understand the use case 100%, but if we want to group by / filter to a specific value of the weight, would it make sense to add a metrics tag "profile" so we can query for example the avg. score for a specific profile?

Yeah I think this would work. Sorry I wasn't really clear about this 😅 . I think we just need a way for us to know what "weight profile" a performance score distribution belongs to so that this can be reflected in the product UI or adjust in some way in the product.

> In that case, Relay could send a 0.5 representing the 50% instead of the raw values and weights.

I haven't thought too much about this, but I _think_ this would work as well, but we would need to do a lot of the calculation back to the score values and weights in post for display in the frontend. I don't expect these weights to change often.

I'm not sure which solution is more efficient here storage wise? But I think I would prefer having some tag(s) on the weights/profiles of the distributions since it might be more flexible and easier to work with.

Reply:
> Since this is still in the experimental phase, imo it's fine to try whichever method since mostly we'll find out quickly if something doesn't make sense once org_id=1 starts using these.

I'm fine with shipping this as-is, but let's have another discussion after the dogfooding phase please.